Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/RideExtra.hs

```diff
@@ -555,174 +558,52 @@ findTotalRidesInDay (Id driverId) time = do
-  pure $ length resTable
-
-totalRidesByFleetOwnerPerVehicleAndDriver :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Maybe Text -> Text -> Id Person -> m Int
-totalRidesByFleetOwnerPerVehicleAndDriver fleetId vehicleNumber driverId = do
-  dbConf <- getMasterBeamConfig
-  res <- L.runDB dbConf $
-    L.findRows $
-      B.select $
-        do
-          rideDetails' <- B.filter_' (\rideDetails'' -> (BeamRD.fleetOwnerId rideDetails'' B.==?. B.val_ fleetId) B.&&?. (BeamRD.vehicleNumber rideDetails'' B.==?. B.val_ vehicleNumber)) (B.all_ (SBC.rideDetails SBC.atlasDB))
-          B.filter_'
-            ( \ride'' -> (ride''.status B.==?. B.val_ DDR.COMPLETED) B.&&?. (BeamR.driverId ride'' B.==?. B.val_ driverId.getId)
-            )
-            do
-              B.join_' (SBC.ride SBC.atlasDB) (\ride'' -> BeamR.id ride'' B.==?. BeamRD.id rideDetails')
-  resTable <- case res of
-    Right res' -> mapM fromTType' res'
-    Left _ -> pure []
-  pure $ length resTable
-
-totalEarningsByFleetOwnerPerVehicleAndDriver :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Maybe Text -> Text -> Id Person -> m Int
-totalEarningsByFleetOwnerPerVehicleAndDriver fleetId vehicleNumber driverId = do
-  dbConf <- getMasterBeamConfig
-  res <-
-    L.runDB dbConf $
-      L.findRows $
-        B.select $
-          do
-            rideDetails' <- B.filter_' (\rideDetails'' -> BeamRD.fleetOwnerId rideDetails'' B.==?. B.val_ fleetId B.&&?. BeamRD.vehicleNumber rideDetails'' B.==?. B.val_ vehicleNumber) (B.all_ (SBC.rideDetails SBC.atlasDB))
-            B.filter_'
-              ( \ride'' -> ride''.status B.==?. B.val_ DDR.COMPLETED B.&&?. BeamR.driverId ride'' B.==?. B.val_ driverId.getId
-              )
-              do
-                B.join_' (SBC.ride SBC.atlasDB) (\ride'' -> BeamR.id ride'' B.==?. BeamRD.id rideDetails')
-  resTable <- case res of
-    Right res' -> mapM fromTType' res'
-    Left _ -> pure []
-  pure $ sum (getMoney <$> mapMaybe DR.fare (catMaybes resTable))
+totalRidesByFleetOwner :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r, CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> m Int
+totalRidesByFleetOwner fleetOwnerId = do
+  rideIds <- findIdsByFleetOwner fleetOwnerId
+  pure $ length rideIds
+
+totalEarningsByFleetOwner :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r, CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> m Int
+totalEarningsByFleetOwner fleetOwnerId = do
+  rideIds <- findIdsByFleetOwner fleetOwnerId
+  getEarningsByIds (cast <$> rideIds)
+
+totalRidesCompletedInFleet :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r, CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> m Int
+totalRidesCompletedInFleet fleetOwnerId = do
+  rideIds <- findIdsByFleetOwner fleetOwnerId
+  getRidesByIdAndStatus (cast <$> rideIds) DRide.COMPLETED
+
+totalRidesCancelledInFleet :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r, CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> m Int
```

Comment:
these three are redundant for clickhouse query
`MonadFlow m, CacheFlow m r, EsqDBFlow m r`